spec/ruby/core/string/append_bytes_spec.rb

```diff
@@ -0,0 +1,28 @@
+require_relative '../../spec_helper'
+
+describe "String#append_bytes" do
+  ruby_version_is "3.4" do
+    it "allow creating broken strings" do
+      str = +"hello"
+      str.append_bytes("\xE2\x82")
+      str.valid_encoding?.should == false
+
+      str.append_bytes("\xAC")
+      str.valid_encoding?.should == true
+    end
+
+    it "never changes the receiver encoding" do
+      str = "".b
+      str.append_bytes("€")
+      str.encoding.should == Encoding::BINARY
+    end
+
```

Comment:
It would probably be good to have a UTF-16 or UTF-32 spec as well.

```ruby
it "handles non-ASCII strings" do
  str = "abc".encode("UTF-16LE")
  str.append_bytes("def")
  str.encoding.should == Encoding::UTF_16LE
  str.valid_encoding?.should == false
end
```